File: src/main/scala/org/jglr/phiengine/network/Message.java
package org.jglr.phiengine.network;

import io.netty.buffer.ByteBuf;
import org.jglr.phiengine.core.PhiEngine;

import java.lang.reflect.InvocationTargetException;

public class Message {

    private final NetworkSide side;
    private final int id;
    public ByteBuf payload;
    public int length;
    public String channel;

    public Message(NetworkSide side, int id) {
        this.side = side;
        this.id = id;
    }

    public int getID() {
        return id;
    }

    public NetworkSide getSide() {
        return side;
    }

    public Packet createPacket() throws InvocationTargetException, NoSuchMethodException, InstantiationException, IllegalAccessException {
        NetworkHandler net = PhiEngine.getInstance().getNetworkHandler();
        return net.newPacket(side, id);
    }
}

File: src/main/scala/org/jglr/phiengine/network/Message.scala
package org.jglr.phiengine.network

import io.netty.buffer.ByteBuf
import org.jglr.phiengine.core.PhiEngine
import java.lang.reflect.InvocationTargetException

import org.jglr.phiengine.network.NetworkSide.NetworkSide
import org.jglr.phiengine.network.channels.NetworkChannel

class Message(val side: NetworkSide, val id: Int) {
  var payload: ByteBuf = null
  var length: Int = 0
  var channel: NetworkChannel = null
  var networkHandler: NetworkHandler = null
  var packetIndex: Long = -1L
  var answerIndex: Long = -1L

  def getID: Int = {
    id
  }

  def getSide: NetworkSide = {
    side
  }

  @throws(classOf[InvocationTargetException])
  @throws(classOf[NoSuchMethodException])
  @throws(classOf[InstantiationException])
  @throws(classOf[IllegalAccessException])
  def createPacket: Packet = {
    val packet = networkHandler.newPacket(side, id)
    packet.packetIndex = packetIndex
    packet.answerIndex = answerIndex
    packet
  }
}